staging/src/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go

```diff
@@ -117,10 +117,25 @@ func WithImpersonation(handler http.Handler, a authorizer.Authorizer, s runtime.
 			}
 		}
 
-		if !groupsSpecified && username != user.Anonymous {
-			// When impersonating a non-anonymous user, if no groups were specified
-			// include the system:authenticated group in the impersonated user info
-			groups = append(groups, user.AllAuthenticated)
+		if username != user.Anonymous {
```

Comment:
I'd like to an `else` on this that ensure that system:anonymous always has `system:unauthenticated` as a group too.

Reply:
@deads2k does not look like we are adding `system:unauthenticated` group for a `system:anonymous` user in the authenticated group adder -  https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/authentication/group/authenticated_group_adder.go#L46. 

I am not sure if we are doing it elsewhere. Do you want me to fix this?

Reply:
I am wrong, it gets added here - https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/authentication/request/anonymous/anonymous.go#L38

Reply:
> I'd like to an `else` on this that ensure that system:anonymous always has `system:unauthenticated` as a group too.

@deads2k  done, please check.